src/OpenLoco/src/Map/TileManager.cpp

```diff
@@ -478,15 +480,36 @@ namespace OpenLoco::World::TileManager
     }
 
     // 0x00462926
-    bool canConstructAt(const World::Pos2& pos, uint8_t baseZ, uint8_t clearZ, const QuarterTile& qt)
+    static bool canConstructAt(const World::Pos2& pos, uint8_t baseZ, uint8_t clearZ, const QuarterTile& qt, void* constructFunction, uint8_t flags)
     {
+        _F00138 = constructFunction != nullptr ? reinterpret_cast<uint32_t>(constructFunction) : 0xFFFFFFFF;
+        _F00166 = flags;
+
         registers regs;
         regs.ax = pos.x;
         regs.cx = pos.y;
         regs.dl = baseZ;
         regs.dh = clearZ;
         regs.bl = qt.getBaseQuarterOccupied() | (qt.getZQuarterOccupied() << 4);
-        return !(call(0x00462926, regs) & Interop::X86_FLAG_CARRY);
+        return !(call(0x00462937, regs) & Interop::X86_FLAG_CARRY);
+    }
+
+    // 0x00462908
+    bool sub_462908(const World::Pos2& pos, uint8_t baseZ, uint8_t clearZ, const QuarterTile& qt, void* constructFunction)
```

Comment:
i would call these clearFunctions not constructFunctions. They only ever remove things. (or do something else :))